torch/nn/modules/rnn.py

```diff
@@ -86,13 +113,49 @@ def forward(self, input, hx=None):
             train=self.training,
             bidirectional=self.bidirectional,
             batch_sizes=batch_sizes,
-            dropout_state=self.dropout_state
+            dropout_state=self.dropout_state,
+            flat_weight=flat_weight
         )
         output, hidden = func(input, self.all_weights, hx)
         if is_packed:
             output = PackedSequence(output, batch_sizes)
         return output, hidden
 
+    def cuda(self, device_id=None):
+        """Moves all model parameters and buffers to the GPU.
```

Comment:
All the .type() functions won't work if RNN is in a module. In this case, _apply would be called for the top level module, which in turn will call _apply on RNN, and never call .type() on RNN directly. Since RNN does not override _apply, the functions below will never be called.